src/gentropy/credible_set_qc.py

```diff
@@ -16,36 +16,42 @@ def __init__(
         self,
         session: Session,
         credible_sets_path: str,
-        study_index_path: str,
-        ld_index_path: str,
         output_path: str,
         p_value_threshold: float = 1e-5,
         purity_min_r2: float = 0.01,
+        clump: bool = False,
+        ld_index_path: str | None = None,
+        study_index_path: str | None = None,
         ld_min_r2: float = 0.8,
     ) -> None:
         """Run credible set quality control step.
 
         Args:
             session (Session): Session object.
             credible_sets_path (str): Path to credible sets file.
-            study_index_path (str): Path to study index file.
-            ld_index_path (str): Path to LD index file.
             output_path (str): Path to write the output file.
             p_value_threshold (float): P-value threshold for credible set quality control. Default is 1e-5.
             purity_min_r2 (float): Minimum R2 for purity estimation. Default is 0.01.
+            clump (bool): Whether to clump the credible sets by LD. Default is False.
+            ld_index_path (str | None): Path to LD index file.
+            study_index_path (str | None): Path to study index file.
             ld_min_r2 (float): Minimum R2 for LD estimation. Default is 0.8.
         """
-        cred_sets = StudyLocus.from_parquet(session, credible_sets_path)
-        study_index = StudyIndex.from_parquet(session, study_index_path)
-        ld_index = LDIndex.from_parquet(session, ld_index_path)
-
+        cred_sets = StudyLocus.from_parquet(
+            session, credible_sets_path, recursiveFileLookup=True
+        ).coalesce(200)
+        if ld_index_path:
+            ld_index =  LDIndex.from_parquet(session, ld_index_path)
+        if study_index_path:
+            study_indedx = StudyIndex.from_parquet(session, study_index_path)
         cred_sets_clean = SUSIE_inf.credible_set_qc(
```

Comment:
@Daniel-Considine I am sorry, I misstyped the suggestion. Now if the `ld_index_path` or `study_index` does not exist, there will be a failure downstream

ps. besides that there is a typo in the `study_indedx`

This would be better :)
```
ld_index = LDIndex.from_parquet(session, ld_index_path) if ld_index_path if ld_index_path else None
```

again sorry for the confusion